handler/handler_test.go

```diff
@@ -688,8 +688,7 @@ func TestFetchFromAnnotationsAPIWithInvalidURL(t *testing.T) {
 
 	assert.Equal(t, http.StatusInternalServerError, resp.StatusCode)
 	assert.NoError(t, err)
-	assert.JSONEq(t, "{\"message\":\"Failed to read annotations: parse :: missing protocol scheme\"}", string(body))
-
+	assert.Contains(t, string(body), "Failed to read annotations")
```

Comment:
Here as well. We can define an error and wrap it. That way we wouldn't degrade the test.

Reply:
Here, we compare strings hence I can't use errors.Is to check for the error type.

Reply:
This is a URL parsing error we can check for *url.Error